crates/aranya-daemon/src/sync/task/quic.rs

```diff
@@ -310,6 +448,198 @@ impl Syncer<State> {
+    ///
+    /// # Returns
+    /// * `Ok(())` if the subscribe request was sent successfully
+    /// * `Err(SyncError)` if there was an error connecting or sending the message
+    #[instrument(skip_all)]
+    async fn send_hello_subscribe_request(
+        &mut self,
+        peer: &Addr,
+        id: GraphId,
+        delay_milliseconds: u64,
+        subscriber_server_addr: Addr,
+    ) -> SyncResult<()> {
+        debug!(
+            ?peer,
+            ?id,
+            delay_milliseconds,
+            ?subscriber_server_addr,
+            "client sending subscribe request to QUIC sync server"
+        );
+
+        // Create the subscribe message
+        let hello_msg = SyncHelloType::Subscribe {
+            delay_milliseconds,
+            address: subscriber_server_addr,
+        };
+        let sync_type: SyncType<Addr> = SyncType::Hello(hello_msg);
+
+        // Serialize the message
+        let data = postcard::to_allocvec(&sync_type).context("postcard serialization failed")?;
+
+        // Connect to the peer
+        let stream = self.connect(peer, id).await?;
+        let (mut recv, mut send) = stream.split();
+
+        // Send the message
+        send.send(Bytes::from(data)).await.map_err(Error::from)?;
+        send.close().await.map_err(Error::from)?;
+
+        // Read the response to avoid race condition with server
+        let mut response_buf = Vec::new();
+        recv.read_to_end(&mut response_buf)
+            .await
+            .context("failed to read hello subscribe response")?;
+        debug!(
+            response_len = response_buf.len(),
+            "received hello subscribe response"
+        );
+
+        debug!("sent subscribe request");
+        Ok(())
+    }
+
+    /// Sends an unsubscribe request to a peer to stop hello notifications.
+    ///
+    /// This method sends a `SyncHelloType::Unsubscribe` message to the specified peer,
+    /// requesting to stop receiving hello notifications when the peer's graph head changes.
+    ///
+    /// # Arguments
+    /// * `peer` - The network address of the peer to send the unsubscribe request to
+    /// * `id` - The graph ID for the team/graph to unsubscribe from
+    /// * `subscriber_server_addr` - The subscriber's server address to identify which subscription to remove
+    ///
+    /// # Returns
+    /// * `Ok(())` if the unsubscribe request was sent successfully
+    /// * `Err(SyncError)` if there was an error connecting or sending the message
+    #[instrument(skip_all)]
+    async fn send_hello_unsubscribe_request(
+        &mut self,
+        peer: &Addr,
+        id: GraphId,
+        subscriber_server_addr: Addr,
+    ) -> SyncResult<()> {
+        debug!("client sending unsubscribe request to QUIC sync server");
+
+        // Create the unsubscribe message
+        let hello_msg = SyncHelloType::Unsubscribe {
+            address: subscriber_server_addr,
+        };
+        let sync_type: SyncType<Addr> = SyncType::Hello(hello_msg);
```

Comment:
`hello_msg` is only used once, so constructing the message could simplify to this:
```suggestion
        let sync_type: SyncType<Addr> = SyncType::Hello(SyncHelloType::Unsubscribe {
            address: subscriber_server_addr,
        });
```